core/src/HIP/Kokkos_HIP_ZeroMemset.hpp

```diff
@@ -29,9 +29,11 @@ void zero_with_hip_kernel(const HIP& exec_space, void* dst, size_t cnt);
 template <>
 struct ZeroMemset<HIP> {
   ZeroMemset(const HIP& exec_space, void* dst, size_t cnt) {
-    // in ROCm <= 6.2.0, hipMemsetAsync on a host-allocated pointer
-    // returns an invalid value error, but accessing the data via a
-    // GPU kernel works.
+    // We allow user on an AMD APU with unified memory to `malloc` and wrap that
+    // in an unmanaged SharedSpace view. In ROCm <= 6.2.1 (and possibly later),
+    // hipMemsetAsync on a host-allocated pointer returns an invalid value
+    // error, but accessing the data via a GPU kernel works as long as xnack is
+    // present and enabled (HSA_XNACK=1)
 #if defined(KOKKOS_IMPL_HIP_UNIFIED_MEMORY)
```

Comment:
I guess either we should make the comment independent of the APU since it also works on older archs.

Reply:
No technically we don't allow this on anything else. From a user perspective HIP_UNIFIED_MEMORY will only ever be on if you have an APU arch.

Reply:
ok, it is only defined for the APU, see https://github.com/kokkos/kokkos/blob/06e5c1303de6c64f20adb0c6dac69b9c2d7e4f91/core/src/setup/Kokkos_Setup_HIP.hpp#L51